### PR TITLE
Multi cohort

### PR DESCRIPTION
**Pull request by 14thibea**
Clean PR for multi-cohort feature addition.

### CODE REVIEW

File: clinicadl/clinicadl/preprocessing/t1_extensive/info.json [hunk 1]
```diff
@@ -4,5 +4,5 @@
     "version": "0.1.0",
     "space_caps": "3M",
     "space_wd": "3M",
-    "dependencies": []
+    "dependencies": [],
```
Comment (mdiazmel) [inline clinicadl/clinicadl/preprocessing/t1_extensive/info.json]: This is an old bug already fixed in the dev branch! (see #115). You have to do `git checkout dev clinicadl/clinicadl/preprocessing/t1_extensive/info.json` for this file. There are maybe other ones...

File: clinicadl/clinicadl/cli.py [hunk 2]
```diff
@@ -330,45 +332,45 @@ def parse_command_line():
...
-    #     help="Probability of each subtype to be drawn in AD label."
-    # )
-    # generate_shepplogan_parser.add_argument(
-    #     '--smoothing',
-    #     action='store_true',
-    #     default=False,
-    #     help='Adds random smoothing to generated data.'
-    # )
-    # generate_shepplogan_parser.set_defaults(func=generate_data_func)
+    generate_shepplogan_parser = generate_subparser.add_parser(
```
Comment (mdiazmel) [inline clinicadl/clinicadl/cli.py]: Just to be sure: do you want to reintroduce the shepplogan generator in this PR or is a merging mistake? (not sure to understand why it's useful in for the multi-cohort feature).

File: clinicadl/clinicadl/cli.py [hunk 3]
```diff
@@ -1090,7 +1098,7 @@ def preprocessing_help(args):
     tsv_getlabels_subparser.add_argument(
         "--diagnoses",
         help="Labels that must be extracted from merged_tsv.",
-        nargs="+", type=str, choices=['AD', 'CN', 'MCI', 'sMCI', 'pMCI'], default=['AD', 'CN'])
+        nargs="+", type=str, choices=['AD', 'BV', 'CN', 'MCI', 'sMCI', 'pMCI'], default=['AD', 'CN'])
```
Comment (mdiazmel) [inline clinicadl/clinicadl/cli.py:1101]: What is about the `BV` label?

File: clinicadl/clinicadl/tools/deep_learning/models/random.py [hunk 4]
```diff
@@ -34,11 +34,10 @@ def random_sampling(rs_options, options):
     """
     Samples all the hyperparameters of the model.
 
-    Args:
-        rs_options: (Namespace) parameters of the random search
-        options: (Namespace) options of the training
+    :param rs_options: (Namespace) parameters of the random search
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/deep_learning/models/random.py]: Another unwanted modification, I suppose... Keep the Google style for the docstrings.

File: clinicadl/clinicadl/tools/deep_learning/models/random.py [hunk 5]
```diff
@@ -113,27 +112,24 @@ def random_sampling(rs_options, options):
 
 def find_evaluation_steps(accumulation_steps, goal=18):
     """
-    Compute the evaluation steps to be a multiple of accumulation steps as close possible as the goal.
+    We want the evaluation step to be performed all goal batches
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/deep_learning/models/random.py]: Same that above!

File: clinicadl/clinicadl/tools/deep_learning/models/random.py [hunk 6]
```diff
@@ -113,27 +112,24 @@ def random_sampling(rs_options, options):
...
         return goal
     else:
-        return (goal // accumulation_steps + 1) * accumulation_steps
+        return find_evaluation_steps(accumulation_steps, goal + 1)
 
 
 def random_conv_sampling(rs_options):
     """
-    Generate random parameters for a random architecture (convolutional part).
+    Generate random parameters for a random architecture (convolutional part)
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/deep_learning/models/random.py]: Same that previous comment.

File: clinicadl/clinicadl/tools/deep_learning/models/random.py [hunk 7]
```diff
@@ -178,13 +174,12 @@ def __init__(self, convolutions, n_fcblocks, initial_shape, dropout=0.5, network
...
         Construct the Architecture randomly chosen for Random Search.
 
-        Args:
-            convolutions: (dict) description of the convolutional blocks.
-            n_fcblocks: (int) number of FC blocks in the network.
-            initial_shape: (list) gives the structure of the input of the network.
-            dropout: (float) rate of the dropout.
-            network_normalization: (str) type of normalization layer in the network.
-            n_classes: (int) Number of output neurones of the network.
+        :param convolutions: (dict) description of the convolutional blocks.
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/deep_learning/models/random.py]: I have the impression that for this file maybe the better will be to `checkout` the `dev` version. To verify!

File: clinicadl/clinicadl/tools/deep_learning/models/slice_level.py [hunk 8]
```diff
@@ -6,6 +6,16 @@
 import math
 from .modules import Flatten
 
+__author__ = "Junhao Wen"
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/deep_learning/models/slice_level.py]: Similar than previous comments. `checkout` the `dev` version for this file.

File: clinicadl/clinicadl/tools/tsv/data_formatting.py [hunk 9]
```diff
@@ -390,6 +411,17 @@ def get_labels(merged_tsv, missing_mods, results_path,
         sub_df = diagnosis_df.reset_index().groupby('participant_id')['session_id'].nunique()
         logger.info('Found %s AD subjects for a total of %s sessions\n' % (len(sub_df), len(diagnosis_df)))
 
+    if 'BV' in diagnoses:
```
Comment (mdiazmel) [inline clinicadl/clinicadl/tools/tsv/data_formatting.py:423]: From here to line 424 are new lines. For the rest of the file, `checkout` the `dev` version.

File: clinicadl/clinicadl/train/random_search.py [hunk 10]
```diff
@@ -17,8 +17,7 @@ def check_and_complete(rs_options):
     This function initializes fields so a random model can be sampled.
     Some fields are mandatory and cannot be initialized by default; this will raise an issue if they are missing.
 
-    Args:
-        rs_options: (Namespace) the random search options
+    :param rs_options: (Namespace) the random search options
```
Comment (mdiazmel) [inline clinicadl/clinicadl/train/random_search.py]: Again, for this file, `checkout` the `dev` version.

**Review by mdiazmel — COMMENTED** [review]
Hi @14thibea ,

Thanks for this nice feature!

I have the impression that something went wrong during the merging of the `upstream/dev` branch. There are some lines of code that bring back previously made modifications. I tried to highlight them in this review (see intermediate comments) but not sure to address all the "bad merging" issues. 

Once this is solved, I would also like to run [`black`](https://github.com/psf/black) and [`isort`](https://github.com/PyCQA/isort) over all the files that you will modify (in order to facilitate the merging with the PR #131. I can do it directly in your fork, if you're agree.

File: clinicadl/clinicadl/cli.py (hunk 3 above)
Comment (14thibea) [reply clinicadl/clinicadl/cli.py:1101]: the BV label exists in NIFD only.
This is not an optimal way to implement new label selection, maybe we could discuss this to improve this point!

File: clinicadl/clinicadl/tools/tsv/data_formatting.py (hunk 9 above)
Comment (14thibea) [reply clinicadl/clinicadl/tools/tsv/data_formatting.py:423]: done

**Review by mdiazmel — APPROVED** [review]
Thanks @14thibea,

Everything seem ok to me, I'll merge this PR!

**Pull request merged**